plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -110,14 +110,34 @@ public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRe
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
             try {
                 final GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
+                String source;
+                boolean found;
                 if (getItemResponse == null || getItemResponse.item() == null || getItemResponse.item().isEmpty()) {
-                    return new GetDataObjectResponse.Builder().id(request.id()).parser(Optional.empty()).build();
+                    found = false;
+                    source = null;
+                } else {
+                    found = true;
+                    source = getItemResponse.item().get(SOURCE).s();
                 }
-
-                String source = getItemResponse.item().get(SOURCE).s();
+                String simulatedGetResponse = "{\"_index\":\""
```

Comment:
This approach tightly couples SDKClient to the opensearch specific schema. This is resulting in non-opensearch clients building mimicking opensearch response which will not be flexible. 

Rather than modifying SDKClient can model group action fetch required field from parser?